### PR TITLE
Changes/ios p2p

### PR DESCRIPTION
**Pull request by mgcouchbase**
IOS app changes for p2p

http://uberjenkins.sc.couchbase.com:8080/view/testkit-testing/job/ios-p2p/11/testReport/
APP build:http://mobile.jenkins.couchbase.com/view/CBL%202.x/job/couchbase-lite-ios-testapp/741/

### CODE REVIEW

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift [hunk 1]
```diff
@@ -95,10 +169,30 @@ public class PeerToPeerRequestHandler {
...
+                let identity = try! TLSIdentity.importIdentity(withData: certData, password: "123", label: "CBL-Cert")
+                replicatorConfig.pinnedServerCertificate = identity.certs[0]
+                print("pinned the certs to Replicator >>>>>>>>>>>>>>>")
+            }
+            if tlsAuthenticator != false {
+                let clientCertData = try dataFromResource(name: "identity/client", ofType: "p12")
+                try! TLSIdentity.deleteIdentity(withLabel: "CBL-Cert")
+                let identity = try TLSIdentity.importIdentity(withData: clientCertData, password: "123", label: "CBL-Cert")
+                replicatorConfig.authenticator = ClientCertificateAuthenticator(identity: identity)
+                print("Added Autheticator to Replicator >>>>>>>>>>>>>>>")
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:191]: when I saw >>>>>>>> I thought your code has git conflicts :) is that possible to use other sign, like ======= or something else?

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift [hunk 2]
```diff
@@ -85,7 +153,13 @@ public class PeerToPeerRequestHandler {
                 }
             }
             
-            let url = URL(string: "ws://\(host):\(port)/\(serverDBName)")!
+            if tls_disable != true {
+               wsPort = "wss"
+            } else {
+                wsPort = "ws"
+            }
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:160]: will this more readable?

if tls_disable {
   wsPort = "ws"
} else {
   wsPort = "wss"
}

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift [hunk 3]
```diff
@@ -47,17 +52,73 @@ public class PeerToPeerRequestHandler {
...
+                print("Setting identity >>>>>>>>>>>>>>>")
+
+            } else if tlsAuthType == "self_signed_create" {
+                try! TLSIdentity.deleteIdentity(withLabel: "CBL-Cert")
+                let id = try! TLSIdentity.createIdentity(forServer: true , attributes: [certAttrCommonName: "CBL-Server"], expiration: nil, label: "CBL-Cert")
+                config.tlsIdentity = id
+                print("Setting CreateIdentity >>>>>>>>>>>>>>")
+            }
+
+             print(args.get(name: "basic_auth")!)
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:100]: I saw there are quite many print statements. Did ios testserver app implement its own logging? java/android has it. I remember I have investigated, just wanna to know

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ListenerAuthenticatorRequestHandler.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            let rootCert = SecCertificateCreateWithData(kCFAllocatorDefault, rootCertData as CFData)!
+
+            let listenerAuth = ListenerCertificateAuthenticator.init(rootCerts: [rootCert])
+            return listenerAuth
+
+        default:
+            throw RequestHandlerError.MethodNotFound(method)
+        }
+
+}
```
Comment (euniceatcouchbase) [inline CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ListenerAuthenticatorRequestHandler.swift:54]: this bracket belongs to func handleRequest, needs tab indent.

**Review by euniceatcouchbase — COMMENTED** [review]
a few minor comments

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ListenerAuthenticatorRequestHandler.swift (hunk 4 above)
Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ListenerAuthenticatorRequestHandler.swift]: done

File: CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift (hunk 2 above)
Comment (mgcouchbase) [reply CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/PeerToPeerRequestHandler.swift:156]: done

**Comment by mgcouchbase** [comment]
Any other comments ?

**Review by euniceatcouchbase — APPROVED** [review]
looks good to me now.

**Review by sridevi-15 — APPROVED** [review]
have same comment as Eunice which I do not see reply . Is print statements generated by xcode automatically as there are lot of print statements?

Also I see jenkins job for p2p which is good. As this is test app repo, it is better to add jenkins job for successful run  for test app too

**Comment by mgcouchbase** [comment]
For Ios we do not have separate logger like android and .net other than print

**Pull request merged**